lib/fluent/config/types.rb

```diff
@@ -20,7 +20,18 @@
 
 module Fluent
   module Config
-    def self.size_value(str)
+    def self.default_value(opts = {})
+      if opts.has_key?(:default)
+        opts[:default]
+      else
+        nil
```

Comment:
Is there a reason why it returns `nil` if the default value is not defined? I think that raising some errors is fine.

Reply:
I agree with you. And now I've noticed that we should handle also `nil` case more properly. The current code causes unhandled error when a plugin doesn't care `nil`.
In addition the current code breaks `Element::to_s`. We should add `nil` case for it.

Reply:
I'm now adding more tests for `Configurable`.
While adding them I've noticed that the implementation for `:default` is wrong.
The default value should be taken from `ConfigProxy::defaults`, not `opts[:default]`.
Otherwise it cannot return a correct default value overridden by `config_set_default`.
I'm now addressing it.

Reply:
I've fixed it.